src/renderer/components/Editor/TakeComponent.tsx

```diff
@@ -87,19 +93,36 @@ const TakeComponent = ({
   isLast,
   isFirstTimeOpen,
   setIsFirstTimeOpen,
+  transcription,
   ...passThroughProps
 }: TakeComponentProps) => {
   const [currentTakeHeight, setCurrentTakeHeight] = useState<number>(0);
   const takeRef = useRef<HTMLDivElement>(null);
 
-  const dispatch = useDispatch();
-
   const onSelectTake = useCallback(() => {
-    dispatch(selectTake(takeWords[0].takeInfo as TakeInfo));
+    const takeInfo = takeWords[0].takeInfo as TakeInfo;
+    const takeGroup = transcription.takeGroups.find(
+      (tg) => tg.id === takeInfo.takeGroupId
+    ) as TakeGroup;
```

Comment:
Can replace cast with an early exit if undefined which is more explicit about edge cases